[Python] Improve "custom instrumentation" docs (#6439)
getsentry/sentry-docs: src/platforms/python/common/performance/instrumentation/custom-instrumentation.mdx

        while pizza.slices > 0:
            with sentry_sdk.start_span(description="Eat Slice"):
                eat_slice(pizza.slices.pop())

```

```python {tabTitle:Decorator}
import sentry_sdk

@sentry_sdk.trace
def eat_slice(slice):
    ...

def eat_pizza(pizza):
    with sentry_sdk.start_transaction(op="task", name="Eat Pizza"):
        while pizza.slices > 0:
            eat_slice(pizza.slices.pop())

```

## Nested Spans

Spans can be nested to form a span tree. (You can read more about that in the [distributed tracing](/product/sentry-basics/tracing/distributed-tracing/) documentation.)

```python {tabTitle:Context Manager}
import sentry_sdk

def chew():
    ...

def swallow():
    ...

def eat_slice(slice):
    with sentry_sdk.start_span(description="Eat Slice"):
        with sentry_sdk.start_span(description="Chew"):
            chew()
        with sentry_sdk.start_span(description="Swallow"):
            swallow()

```

```python {tabTitle:Decorator}
import sentry_sdk

@sentry_sdk.trace
def chew():
    ...

@sentry_sdk.trace
def swallow():
    ...

@sentry_sdk.trace
def eat_slice(slice):
    chew()
    swallow()
```

## Accessing the Current Transaction

> In cases where you want to change data in an already ongoing Transaction you can use `Hub.current.scope.transaction`. This property will return a Transaction in case there is a running Transaction otherwise it returns `None`.
> 

Reviewer: ```suggestion
To change data in an already ongoing transaction, use `Hub.current.scope.transaction`. This property will return a transaction if there's one running, otherwise it will return `None`.

```

```python
import sentry_sdk

def eat_pizza(pizza):
    transaction = sentry_sdk.Hub.current.scope.transaction

    if transaction is not None:
        transaction.set_tag("num_of_slices", len(pizza.slices))

    while pizza.slices > 0:
        eat_slice(pizza.slices.pop())
```

## Accessing the Current Span

> 
> In cases where you want to change data in the current Span you can use `sentry_sdk.Hub.current.scope.span`. This property will return a Span in case there is a running Span otherwise it returns `None`.
> 
> In this example we will set a tag in the Span created by the `@sentry_sdk.trace` decorator.
> 

Reviewer: ```suggestion

To change data in the current span, use `sentry_sdk.Hub.current.scope.span`. This property will return a span if there's one running, otherwise it will return `None`.

In this example, we'll set a tag in the span created by the `@sentry_sdk.trace` decorator.

```
